src/main/java/mate/academy/bookstore/serivce/category/CategoryServiceImpl.java

```
@@ -0,0 +1,61 @@
+import mate.academy.bookstore.mapper.CategoryMapper;
+import mate.academy.bookstore.model.Category;
+import mate.academy.bookstore.repository.CategoryRepository;
+import mate.academy.bookstore.serivce.book.BookService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookService bookService;
+
+    @Override
+    public CategoryResponseDto save(CreateCategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public List<CategoryResponseDto> findAll(Pageable pageable) {
+        List<Category> categoryList = categoryRepository.findAll(pageable).stream().toList();
+        return categoryMapper.toDtoList(categoryList);
+    }
+
+    @Override
+    public CategoryResponseDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Category with id " + id + " not found"));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long id, Pageable pageable) {
+        return bookService.getBooksByCategoryId(id, pageable);
+    }
+
+    @Override
+    public CategoryResponseDto updateById(Long id, CreateCategoryRequestDto requestDto) {
+        if (categoryRepository.existsById(id)) {
+            Category category = categoryMapper.toModel(requestDto);
+            category.setId(id);
+            return categoryMapper.toDto(categoryRepository.save(category));
+        }
+        throw new EntityNotFoundException("Category with id " + id + " not found");
```

reverse condition to fail fast